Update diagram for ALTER EXTERNAL CONNECTION

peachdawnleach opened the pull request:
docgen: update ALTER EXTERNAL CONNECTION diagram
Epic: none
Release note: none
Release justification: non-production code change

Part of [DOC-13254](https://cockroachlabs.atlassian.net/browse/DOC-13254)

New diagram:
<img width="1023" height="283" alt="Screenshot 2025-10-17 at 7 24 43 PM" src="https://github.com/user-attachments/assets/21975cb9-e9ef-4162-8782-7249436bad05" />

peachdawnleach committed:
Updated diagram

Updated SQL railroad diagram for ALTER EXTERNAL CONNECTION with correct parameter names

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/155664)
<!-- Reviewable:end -->

peachdawnleach committed:
Delete alter_external_connection_stmt.bnf

This file was generated prior to my updates and should be removed, as the updated file no longer contains '_stmt' in the name

peachdawnleach commented:
bors r=kev-cao
